CMakeLists.txt

```diff
@@ -37,13 +38,13 @@ endif ()
 if (UNIX)
 	
 	set(CMAKE_SHARED_LIBRARY_PREFIX )
-	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -s -Wl,--no-undefined")
 
 	add_compile_options("-Wall")
-	add_compile_options("-fvisibility=hidden")
 	add_compile_options("-fvisibility-inlines-hidden")
-	# add_link_options("-s")
-	# add_link_options("-Wl,--no-undefined")
+
+  if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    add_link_options("LINKER:--no-undefined")
```

Comment:
This is not the same as before (missing `-s`, and potentially `-Wl` - I guess that the latter is replaced by `LINKER:`?)

`add_link_options` is not support by the CMake version that comes with Ubuntu 18.04. I guess that the GitHub image has a more recent version, but it will not work on a local computer with a standard Ubuntu install. Hence why it was commented out before.

Reply:
`-s` is not portable and is used to strip the symbols. We are now using the `strip` command in a post build step instead. I confirm that the output of `-s` and `strip` is identical on a vanilla Ubuntu 18.04.

Good point about `add_link_options`, I reverted it in the latest commit.